resources/files/runtime/runtime.sh

```diff
@@ -0,0 +1,51 @@
+#!/bin/bash
+
+# Purpose: copy over shared object files for libgcc, libssp and libstdc++ into
+# the lib directory of the puppet agent. The files and links are determined
+# dynamically because different versions of GCC produce different library
+# versions and filenames. This script should apply cleanly on any Linux that
+# requires a runtime setup for puppet-agent builds. If the libraries are being
+# linked in statically, this script will be a no-op.
+
+set -e
+
+if [ -z "$1" ] ; then
+  LIBDIR=/opt/pl-build-tools/lib
+else
+  if [ -d "$1" ] ; then
+    LIBDIR="$1"
+  else
+    echo "$1 does not exist"
+    exit 1
+  fi
+fi
+
+RUNTIMEDIR=/opt/puppetlabs/puppet/lib
+
+mkdir -p "$RUNTIMEDIR"
+
+# Find the .so files we need
+# shellcheck disable=SC2045
+for filepath in $(ls "$LIBDIR"/libgcc* "$LIBDIR"/libssp* "$LIBDIR"/libstdc++*)
```

Comment:
Yeaaaaaaah but parsing `ls` output is dicey and I'd rather not run `set -e` than parse `ls` output